### PR TITLE
End to End SQL VM AG, AG listener setup

### PR DESCRIPTION
**Pull request by shiva08**

### CODE REVIEW

File: AzureSQLVM/e2e-ag-setup/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+4. Subnets for VMs
+
+## Deployment Steps
+
+You can click the "deploy to Azure" button at the beginning of this document or follow the instructions for command line deployment using the scripts in the root of this repo.
+
+## Usage
+
+Template can be deployed with multiple clients : Portal, CLI, Powershell, Rest API, Cloud Shell.
+
```
Comment (bluefooted) [inline AzureSQLVM/e2e-ag-setup/README.md:49]: Can we add something here that talks about how this template creates an AG that adheres to our recommended best practices as documented at https://docs.microsoft.com/azure/azure-sql/virtual-machines/windows/hadr-cluster-best-practices?view=azuresql&tabs=windows2012

**Review by bluefooted — CHANGES_REQUESTED** [review]

File: AzureSQLVM/e2e-ag-setup/README.md (hunk 1 above)
Comment (shiva08) [reply AzureSQLVM/e2e-ag-setup/README.md]: @bluefooted  I added some information. Can you check now?

File: AzureSQLVM/e2e-ag-setup/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+4. Creates Storage account (if it doesn't exist already) which will act as Witness for Failover Cluster
+5. Creates Windows server Failover cluster
+6. Runs necessary checks such as TEST Cluster to ensure the health of creation of cluster 
+7. Creates Availability Group 
+8. Creates Availability Group Listener
+
+Best practices that were considered for this solution:
+1. This solution deploys SQL Server VMs to multiple subnets,thereby avoiding the dependency on an Azure Load Balancer or a distributed network name (DNN) to route traffic to your HADR solution.
+2. Use a single NIC per cluster node.
+
```
Comment (bluefooted) [inline AzureSQLVM/e2e-ag-setup/README.md:24]: Sorry I was thinking that all of the best practices were implemented so I just wanted to include the link. I think it's better to just remove this section and not say anything about best practices until we have them all implemented.

**Review by bluefooted — CHANGES_REQUESTED** [review]

File: AzureSQLVM/e2e-ag-setup/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Before deploying the template you must have the following:
+1. A Virtual Network 
+2. A domain controller VM in the same virtual network
+3. Accounts 
+    SQL service
+    DomainAdmin - This should have Create Computer object permissions.
+4. Subnets for VMs. Refer https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-manage-subnet#add-a-subnet
+
+Notes: 
+1. This [template] (https://github.com/Azure/azure-quickstart-templates/tree/master/application-workloads/active-directory/active-directory-new-domain) is helpful for above steps 1,2,3
```
[the remark this answers is not in the document]
Comment (shiva08) [reply AzureSQLVM/e2e-ag-setup/README.md:48]: Specified the prerequsite steps

File: AzureSQLVM/e2e-ag-setup/README.md (hunk 2 above)
Comment (shiva08) [reply AzureSQLVM/e2e-ag-setup/README.md:24]: Ok, I will remove this section

**Comment by bluefooted** [comment]
Thank you @shiva08 !!

**Pull request merged**